From: Tommy Huang <tommy_huang@aspeedtech.com>
Date: Mon, 9 Mar 2026 10:38:35 +0800
Subject: [PATCH 6.12 1/1] i2c: ast2600: change the remove driver call back function

Upstream-Status: Backport [AspeedTech-BMC/linux aspeed-master-v6.12 d3c1f3c535637161129b4202fe5d3c55946e60fa, SDK v00.07.03]

Change back the remove driver call back function from
remove_new to remove

Signed-off-by: Tommy Huang <tommy_huang@aspeedtech.com>
Change-Id: I3b56757cd5f11b7d5365238dcb8e58f6f83434c5

[hw-mgmt] Not part of MR !1037. Included so the Aspeed v00.07.02..v00.07.04
i2c-ast2600 series applies unmodified; .remove and .remove_new are
equivalent on 6.12.
---
 drivers/i2c/busses/i2c-ast2600.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-ast2600.c b/drivers/i2c/busses/i2c-ast2600.c
index e809d82d9..91117032a 100644
--- a/drivers/i2c/busses/i2c-ast2600.c
+++ b/drivers/i2c/busses/i2c-ast2600.c
@@ -2435,7 +2435,7 @@ MODULE_DEVICE_TABLE(of, ast2600_i2c_bus_of_table);
 
 static struct platform_driver ast2600_i2c_bus_driver = {
 	.probe = ast2600_i2c_probe,
-	.remove_new = ast2600_i2c_remove,
+	.remove = ast2600_i2c_remove,
 	.driver = {
 		.name = KBUILD_MODNAME,
 		.of_match_table = ast2600_i2c_bus_of_table,
